tests/testthat/test-pmcmc.R

```diff
@@ -350,63 +280,76 @@ test_that("pmcmc with new model", {
   
   ## test generalised version
   
-  pars_to_sample <- c('beta_start','beta_end', 'start_date',  'gamma_triage', 'gamma_hosp_R', 
-                      'gamma_hosp_D', 'gamma_ICU_R', 'gamma_ICU_D', 'gamma_stepdown')
-  
-  proposal_kernel <- diag(length(pars_to_sample)) * 0.01^2
-  row.names(proposal_kernel) <- colnames(proposal_kernel) <- pars_to_sample
+  pars_to_sample <- data.frame(
+    names=c('beta_start',
+            'beta_end', 
+            'start_date',  
+            'gamma_triage', 
+            'gamma_hosp_R', 
+            'gamma_hosp_D', 
+            'gamma_ICU_R', 
+            'gamma_ICU_D', 
+            'gamma_stepdown'),
+    init=c(0.14, 
+           0.14*0.238,
+           as.Date("2020-02-07"),
+           0.5099579,
+           0.1092046,
+           0.2911154,
+           0.3541429,
+           0.2913861,
+           0.452381),
+    min=c(0,
+          0,
+          0,
+          0,
+          0,
+          0,
+          0,
+          0,
+          0),
+    max=c(1,
+          1,
+          1e6,
+          1,
+          1,
+          1,
+          1,
+          1,
+          1),
+    discrete=c(FALSE,
+               FALSE,
+               TRUE,
+               FALSE,
+               FALSE,
+               FALSE,
+               FALSE,
+               FALSE,
+               FALSE),
+    stringsAsFactors = FALSE)
+  pars_lprior <- list('beta_start'     = function(pars) log(1e-10),
+                     'beta_end'       = function(pars) 0,
+                     'start_date'     = function(pars) 0,
+                     'gamma_triage'   = function(pars) 0,
+                     'gamma_hosp_R'   = function(pars) 0,
+                     'gamma_hosp_D'   = function(pars) 0,
+                     'gamma_ICU_R'    = function(pars) 0,
+                     'gamma_ICU_D'    = function(pars) 0,
+                     'gamma_stepdown' = function(pars) 0)
+  
+  proposal_kernel <- diag(length(pars_to_sample$names)) * 0.01^2
+  row.names(proposal_kernel) <- colnames(proposal_kernel) <- pars_to_sample$names
```

Comment:
Could we add a test that checks the output is robust to the order in which the parameters are entered into pars_to_sample. I'm thinking specifically wrt the proposal kernel?

Reply:
Great idea, this is indeed a bug in the current version